src/EcommerceStatsBundle/Service/TopSellerService.php

```diff
@@ -0,0 +1,84 @@
+<?php declare(strict_types=1);
+
+namespace ChameleonSystem\EcommerceStatsBundle\Service;
+
+use ChameleonSystem\EcommerceStatsBundle\DataModel\ShopOrderItemDataModel;
+use ChameleonSystem\EcommerceStatsBundle\Interfaces\TopSellerServiceInterface;
+use Doctrine\DBAL\Connection;
+
+class TopSellerService implements TopSellerServiceInterface
+{
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    /**
+     * @return ShopOrderItemDataModel[]
+     */
+    public function getTopsellers(
+        ?string $startDate,
+        ?string $endDate,
+        string $portalId,
+        int $limit = 50
+    ): array {
+        $query = '
+            SELECT 
+                SUM(`shop_order_item`.`order_amount`) AS totalordered,
+                SUM(`shop_order_item`.`order_price_after_discounts`) AS totalorderedvalue,
+                `shop_category`.`url_path` AS categorypath, 
+                `shop_order_item`.*
+            FROM `shop_order_item`
+                LEFT JOIN `shop_order`
```

Comment:
left join: Could there be then entries without order data?

Reply:
I don't think there could be entries without order data - but I would probably still argue that it's safer to leave it as a `LEFT JOIN` just to be sure